test/conftest.py

```diff
@@ -48,3 +48,13 @@ def _getkey():
         return jr.PRNGKey(random.randint(0, 2**31 - 1))
 
     return _getkey
+
+
+@pytest.fixture(scope="module")
+def beartype_or_skip():
+    yield pytest.importorskip("beartype")
+
+
+@pytest.fixture(scope="module")
```

Comment:
IIUC, making these module-level fixtures will result in a whole file getting skipped if these aren't present?

Reply:
No, not exactly: if beartype or typeguard are not found, only the tests that use this fixture will be skipped. For example, if I uninstall beartype I get this report after doing ```pytest```:
```56 passed, 30 skipped, 1 warning in 5.40s```.
Or, when doing ```pytest test/test_import_hook```: ```4 passed, 3 skipped in 0.14s```.

So, I think it should work fine. At some point I would also like to change the codebase to support adequately either only typeguard or beartype, but this is not for this PR.